File: Tech Module/Software Technologies/Exam_Preparation_03/IMDB PHP/src/AppBundle/Controller/FilmController.php
<?php

namespace AppBundle\Controller;

use AppBundle\Entity\Film;
use AppBundle\Form\FilmType;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
use Symfony\Bundle\FrameworkBundle\Controller\Controller;
use Symfony\Component\HttpFoundation\Request;

class FilmController extends Controller
{
    /**
     * @param Request $request
     * @Route("/", name="index")
     * @return \Symfony\Component\HttpFoundation\Response
     */
    public function index(Request $request)
    {
        $allFilms = $this
            ->getDoctrine()
            ->getRepository('AppBundle:Film')
            ->findAll();

        return $this->render(':film:index.html.twig', ['films' => $allFilms]);
    }

    /**
     * @param Request $request
     * @Route("/create", name="create")
     * @return \Symfony\Component\HttpFoundation\Response
     */
    public function create(Request $request)
    {
        $filmToCreate = new Film();

        $form = $this->createForm(FilmType::class, $filmToCreate);

        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {

            if ($filmToCreate->getName() == null
                || $filmToCreate->getGenre() == null
                || $filmToCreate->getDirector() == null
                || $filmToCreate->getYear() == null) {
                return $this->redirectToRoute('index');
            }

            if ($filmToCreate->getName() != ""
                && $filmToCreate->getGenre() != ""
                && $filmToCreate->getDirector() != ""
                && $filmToCreate->getYear() != "") {

                $em = $this->getDoctrine()->getManager();
                $em->persist($filmToCreate);
                $em->flush();
            }

            return $this->redirectToRoute('index');
        }

        return $this->render('film/create.html.twig', ['form' => $form->createView()]);
	}

    /**
     * @Route("/edit/{id}", name="edit")
     *
     * @param $id
     * @param Request $request
     * @return \Symfony\Component\HttpFoundation\Response
     */
    public function edit($id, Request $request)
    {
        $filmToEdit = $this->getDoctrine()->getRepository(Film::class)->find($id);

        if ($filmToEdit == null) {
            return $this->redirectToRoute('index');
        }

        $form = $this->createForm(FilmType::class, $filmToEdit);

        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {

            $em = $this->getDoctrine()->getManager();
            $em->merge($filmToEdit);
            $em->flush();

            return $this->redirectToRoute('index');
        }

        return $this->render('film/edit.html.twig', ['film' => $filmToEdit, 'form' => $form->createView()]);
    }

    /**
     * @Route("/delete/{id}", name="delete")
     *
     * @param $id
     * @param Request $request
     * @return \Symfony\Component\HttpFoundation\Response
     */
    public function delete($id, Request $request)
    {
        $filmToDelete = $this->getDoctrine()->getRepository(Film::class)->find($id);

        if ($filmToDelete == null) {
            return $this->redirectToRoute('index');
        }

        $form = $this->createForm(FilmType::class, $filmToDelete);

        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {

            $em = $this->getDoctrine()->getManager();
            $em->remove($filmToDelete);
            $em->flush();

            return $this->redirectToRoute('index');
        }

        return $this->render('film/delete.html.twig', ['film' => $filmToDelete, 'form' => $form->createView()]);
    }
}

File: Tech Module/Software Technologies/Exam_Preparation_03/IMDB CSharp/IMDB/Controllers/FilmController.cs
using System.Linq;
using System.Net;
using System.Web.Mvc;
using IMDB.Models;

namespace IMDB.Controllers
{
    [ValidateInput(false)]
    public class FilmController : Controller
    {
        [HttpGet]
        [Route("")]
        public ActionResult Index()
        {
            using (var db = new IMDBDbContext())
            {
                var allFilms = db.Films.ToList();

                return View(allFilms);
            }
        }

        [HttpGet]
        [Route("create")]
        public ActionResult Create()
        {
            return View();
        }

        [HttpPost]
        [Route("create")]
        [ValidateAntiForgeryToken]
        public ActionResult Create(Film filmToCreate)
        {
            if (filmToCreate == null)
            {
                return RedirectToAction("Index");
            }

            if (string.IsNullOrWhiteSpace(filmToCreate.Name) ||
                string.IsNullOrWhiteSpace(filmToCreate.Genre) ||
                string.IsNullOrWhiteSpace(filmToCreate.Director))
            {
                return RedirectToAction("Index");
            }

            using (var db = new IMDBDbContext())
            {
                db.Films.Add(filmToCreate);
                db.SaveChanges();

                return RedirectToAction("Index");
            }
        }

        [HttpGet]
        [Route("edit/{id}")]
        public ActionResult Edit(int? id)
        {
            if (id == null)
            {
                return RedirectToAction("Index");
            }

            using (var db = new IMDBDbContext())
            {
                var filmToEdit = db.Films.Find(id);

                if (filmToEdit == null)
                {
                    return RedirectToAction("Index");
                }

                return View(filmToEdit);
            }
        }

        [HttpPost]
        [Route("edit/{id}")]
        [ValidateAntiForgeryToken]
        public ActionResult EditConfirm(int? id, Film filmModel)
        {
            if (id == null)
            {
                return RedirectToAction("Index");
            }

            using (var db = new IMDBDbContext())
            {
                var filmToEdit = db.Films.Find(id);

                if (filmToEdit == null)
                {
                    return RedirectToAction("Index");
                }

                if (string.IsNullOrWhiteSpace(filmToEdit.Name) ||
                string.IsNullOrWhiteSpace(filmToEdit.Genre) ||
                string.IsNullOrWhiteSpace(filmToEdit.Director))
                {
                    return RedirectToAction("Index");
                }

                db.Films.Find(id).Name = filmModel.Name;
                db.Films.Find(id).Genre = filmModel.Genre;
                db.Films.Find(id).Director = filmModel.Director;
                db.Films.Find(id).Year = filmModel.Year;

                db.SaveChanges();

                return RedirectToAction("Index");
            }
        }

        [HttpGet]
        [Route("delete/{id}")]
        public ActionResult Delete(int? id)
        {
            if (id == null)
            {
                return RedirectToAction("Index");
            }

            using (var db = new IMDBDbContext())
            {
                var filmToDelete = db.Films.Find(id);

                if (filmToDelete == null)
                {
                    return RedirectToAction("Index");
                }

                return View(filmToDelete);
            }
        }

        [HttpPost]
        [Route("delete/{id}")]
        [ValidateAntiForgeryToken]
        public ActionResult DeleteConfirm(int? id)
        {
            if (id == null)
            {
                return RedirectToAction("Index");
            }

            using (var db = new IMDBDbContext())
            {
                var filmToDelete = db.Films.Find(id);

                if (filmToDelete == null)
                {
                    return RedirectToAction("Index");
                }

                db.Films.Remove(filmToDelete);
                db.SaveChanges();

                return RedirectToAction("Index");
            }
            
        }
    }
}